Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -411,39 +411,49 @@ public void SetCash(decimal cash)
         ///     Similarly the desired trade direction can impact the buying power available
         /// </remarks>
         /// <returns>Decimal total buying power for this symbol</returns>
-        public decimal GetBuyingPower(string symbol, OrderDirection direction = OrderDirection.Hold) 
+        public decimal GetFreeCash(string symbol, OrderDirection direction = OrderDirection.Hold) 
         {
             //Each asset has different leverage values, so affects our cash position in different ways.
-            // Basically position affect on cash = holdings / leverage
-            var holdings = (Securities[symbol].Holdings.AbsoluteQuantity * Securities[symbol].Close);
-            var remainingBuyingPower = Cash * Securities[symbol].Leverage;
+            var holdings = Securities[symbol].Holdings;
+            var remainingCash = Cash;
 
-            if (direction == OrderDirection.Hold) return remainingBuyingPower;
+            var price = Securities[symbol].Price;
+            var leverage = Securities[symbol].Leverage;
+
+            if (direction == OrderDirection.Hold) return remainingCash;
             //Log.Debug("SecurityPortfolioManager.GetBuyingPower(): Direction: " + direction.ToString());
 
+
+            //If the order is in the same direction as holdings, our remaining cash is our cash
+            //In the opposite direction, our remaining cash is 2 x current value of assets + our cash
             if (Securities[symbol].Holdings.IsLong)
             {
                 switch (direction)
                 {
                     case OrderDirection.Buy:
-                        return remainingBuyingPower;
+                        return remainingCash;
                     case OrderDirection.Sell:
-                        return holdings * 2  + remainingBuyingPower;
+                        var profit = (price - holdings.AveragePrice) * holdings.AbsoluteQuantity;
```

Comment:
Excellent as usual @bdilber; perhaps to centralize the formulas we could use the holdings class here to get the unrealized profit?
https://github.com/QuantConnect/Lean/blob/master/Common/Securities/SecurityHolding.cs#L293